### PR TITLE
Implement adaptive header

### PR DESCRIPTION
**Pull request by vladimir-vladykin**
Implementation of header, plus add reusable parts into base.css (logo, navigation links, socials)

### CODE REVIEW

File: src/css/base/base.css [hunk 1]
```diff
@@ -111,3 +111,80 @@ main:has(.back-link) {
...
+  flex-direction: row;
+  align-items: center;
+  gap: 12px;
+}
+.main-socials__link {
+  /* apply alpha channel to black color to achieve border at 10% */
+  border: 1px solid rgb(from currentColor r g b / 0.1);
+  border-radius: 10px;
+  padding: 7px; /* note padding here, it's 7 instead of 8, to consider 1px border and preserve total size of item as 44px */
+
```
Comment (ElephantFrontend) [inline src/css/base/base.css:168]: formatted code please (spaces)

File: src/css/base/base.css [hunk 2]
```diff
@@ -111,3 +111,80 @@ main:has(.back-link) {
...
+/* #region Socials */
+/* Socials, reusable part */
+.main-socials__list {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  gap: 12px;
+}
+.main-socials__link {
+  /* apply alpha channel to black color to achieve border at 10% */
```
Comment (ElephantFrontend) [inline src/css/base/base.css]: remove please so many comments =)

File: src/css/components/header.css [hunk 3]
```diff
@@ -0,0 +1,110 @@
+.header {
+  /* to achieve mobile header height 68px */
+  padding: 18px 0px;
+}
+@media only screen and (min-width: 768px) {
```
Comment (ElephantFrontend) [inline src/css/components/header.css]: all media please to bottom
we need only 2 media

File: src/css/components/header.css [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+.header__container {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+}
+
+.header__mobile_menu_button {
```
Comment (ElephantFrontend) [inline src/css/components/header.css]: please rename to .menu-button

File: src/css/components/header.css [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  }
+}
+@media only screen and (min-width: 1280px) {
+  .header {
+    /* to achieve desktop height 84px */
+    padding: 20px 0px;
+  }
+}
+
+.header__container {
```
Comment (ElephantFrontend) [inline src/css/components/header.css]: please don't add one more class for container. Use .header .container

File: src/css/base/base.css [hunk 6]
```diff
@@ -111,3 +111,80 @@ main:has(.back-link) {
...
+.logo {
+  color: var(--green-color);
+  font-weight: 700;
+  font-size: 20px;
+}
+/* #endregion */
+
+/* #region Nav links */
+/* Navigation links, reusable part */
+.nav_link {
```
Comment (ElephantFrontend) [inline src/css/base/base.css:126]: if it is main menu.
We need to styling in header.css

File: src/partials/header.html [hunk 7]
```diff
@@ -0,0 +1,84 @@
+<!-- Adaptive header, by Volodymyr Vladykin -->
+<header class="header">
+  <div class="header__container container container__content">
+    <!-- Logo, link to core page -->
```
Comment (ElephantFrontend) [inline src/partials/header.html]: don't need so many comments =)

File: src/partials/header.html [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+      </svg>
+    </button>
+
+    <!-- Main navigation -->
+    <nav class="nav">
+      <!-- Navigation menu (visible only on desktop) -->
+      <ul class="header__nav_list">
+        <li>
+          <a href="#" class="nav_link header__nav_link--dark">About us</a>
+        </li>
```
Comment (ElephantFrontend) [inline src/partials/header.html:24]: formatted code please

File: src/partials/header.html [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+    <!-- Mobile menu button, will be hidden on desktop -->
+    <button class="header__mobile_menu_button" type="button">
+      <svg class="header__mobile_menu_icon" width="32" height="32">
+        <use href="./img/sprite.svg#icon-menu"></use>
+      </svg>
+    </button>
+
+    <!-- Main navigation -->
+    <nav class="nav">
```
Comment (ElephantFrontend) [inline src/partials/header.html]: change class name please

File: src/css/base/base.css [hunk 10]
```diff
@@ -111,3 +111,80 @@ main:has(.back-link) {
...
+  /* to power pseudo-element */
+  position: relative;
+
+  /* just for case, since size of pseudo-element is 37px */
+  min-width: 37px;
```
Comment (ElephantFrontend) [inline src/css/base/base.css]: we don't need min-width

File: src/css/base/base.css (hunk 1 above)
Comment (vladimir-vladykin) [reply src/css/base/base.css:168]: Not clear what exactly wrong with spaces here
Also, shouldn't this kind of formatting just be enforced via .prettierrc.json ?

File: src/css/base/base.css (hunk 6 above)
Comment (vladimir-vladykin) [reply src/css/base/base.css:126]: The same links will be reused in footer, based on design from figma, the same text appearance, same hover, the only difference between links in header and footer is text color

File: src/partials/header.html (hunk 8 above)
Comment (vladimir-vladykin) [reply src/partials/header.html:24]: Not sure what wrong here with formatting, this is result of Prettier formatting

File: src/css/base/base.css (hunk 1 above)
Comment (ElephantFrontend) [reply src/css/base/base.css:168]: correct 
.class {
  margin: 0;
  padding: 0;
  height: 10px;
}

.class2 {
  margin: 0;
}

incorrect:
.class {
  margin: 0;
  
  padding: 0;
  height: 10px;
}
.class2 {
  margin: 0;
}

File: src/css/base/base.css (hunk 6 above)
Comment (ElephantFrontend) [reply src/css/base/base.css:126]: Ok

File: src/partials/header.html (hunk 8 above)
Comment (ElephantFrontend) [reply src/partials/header.html:24]: Ok
We need to have call for check

File: src/css/base/base.css [hunk 11]
```diff
@@ -111,3 +111,75 @@ main:has(.back-link) {
     height: 20px;
   }
 }
+
+/* #region Logo */
```
Comment (ElephantFrontend) [inline src/css/base/base.css:115]: please remove so many comments

Comment (ElephantFrontend) [reply src/css/base/base.css:115]: will fixed when we will check all project

File: src/css/base/base.css (hunk 1 above)
Comment (ElephantFrontend) [reply src/css/base/base.css:168]: will fixed when we will check all project

File: src/partials/header.html (hunk 8 above)
Comment (ElephantFrontend) [reply src/partials/header.html:24]: will fixed when we will check all project

**Pull request merged**